### PR TITLE
DOC: Multiindex sort docs

### LINKED ISSUE #13108: DOC: MultiIndex sort docs
I found this confusing, despite being a moderately competent pandas user:

from http://pandas.pydata.org/pandas-docs/stable/advanced.html#the-need-for-sortedness-with-multiindex

>Caveat emptor: the present implementation of MultiIndex requires that the labels be sorted for some of the slicing / indexing routines to work correctly. You can think about breaking the axis into unique groups, where at the hierarchical level of interest, each distinct group shares a label, but no two have the same label. However, the MultiIndex does not enforce this: you are responsible for ensuring that things are properly sorted. There is an important new method sort_index to sort an axis within a MultiIndex so that its labels are grouped and sorted by the original ordering of the associated factor at that level. Note that this does not necessarily mean the labels will be sorted lexicographically!

Is this right, that calling `sort_index` doesn't guarantee lex sortedness? How to guarantee it then?

And this:

>Some indexing will work even if the data are not sorted, but will be rather inefficient and will also return a copy of the data rather than a view:

...seems to contradict this:

>Thus, if you try to index at a depth at which the index is not sorted, it will raise an exception.

...neither of which seems tightly consistent with the passage above.

Am I misunderstanding something?

### PR DESCRIPTION
**Pull request by max-sixty**
closes https://github.com/pydata/pandas/issues/13108

### CODE REVIEW

File: doc/source/advanced.rst [hunk 1]
```diff
@@ -510,24 +503,23 @@ are named.
...
 .. ipython:: python
 
-   s['qux']
-   s.sort_index(level=1)['qux']
+   df.T.sort_index(level=1, axis=1)
 
-On higher dimensional objects, you can sort any of the other axes by level if
-they have a MultiIndex:
+Some indexing will work even if the data are not sorted, but will be rather
+inefficient (and raises a ``PerformanceWarning``). It will will also
```
Comment (jreback) [inline doc/source/advanced.rst]: 1 will

shows a warning (not raises)

**Comment by MaximilianR** [comment]
@jreback updated

**Comment by jreback** [comment]
change around a little bit. thanks for the PR!

File: doc/source/advanced.rst [hunk 2]
```diff
@@ -538,18 +530,6 @@ an exception. Here is a concrete example to illustrate this:
...
-
-However:
-
-::
-
-   >>> s.ix[('a', 'b'):('b', 'a')]
-   Traceback (most recent call last)
-        ...
-   KeyError: Key length (3) was greater than MultiIndex lexsort depth (2)
-
```
Comment (jorisvandenbossche) [inline doc/source/advanced.rst]: Is there a reason this is removed? I think it is still a nice example of what happens if the index is not sorted? (this error still occurs on master)

Comment (jreback) [inline doc/source/advanced.rst]: I added it back (in the context of the slightly changed example). lmk if more clear.
cc @MaximilianR

Comment (jreback) [inline doc/source/advanced.rst]: https://github.com/pydata/pandas/commit/43989fd7cb9917e885c2b55a172c4f9f3838d59d

**Pull request closed without merging**